src/client/components/DicePanel.tsx

```diff
@@ -1,57 +1,382 @@
 import {
-  faBalanceScale,
-  faBroom,
   faDiceD20,
-  faDragon,
-  faFireAlt,
-  faMagic,
+  faHandPaper,
+  faPlus,
   faShieldAlt,
 } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import clsx from "clsx";
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
+import {
+  playerAddDiceTemplateCategory,
+  playerUpdateDiceTemplateCategory,
+  playerDeleteDiceTemplateCategory,
+} from "../../shared/actions";
+import { DEFAULT_SYNC_TO_SERVER_DEBOUNCE_TIME } from "../../shared/constants";
+import {
+  categoryIcons,
+  characterStatNames,
+  fixedCategoryIcons,
+  iconMap,
+  RRDiceTemplatePart,
+  RRDiceTemplatePartDice,
+  skillMap,
+  skillNames,
+  userCategoryIcons,
+} from "../../shared/state";
+import { rrid } from "../../shared/util";
+import {
+  RRDiceTemplate,
+  RRDiceTemplateCategory,
+} from "../../shared/validation";
+import { useMyProps } from "../myself";
+import { useServerDispatch, useServerState } from "../state";
 import { DiceInput } from "./DiceInput";
 import { DiceInterface } from "./DiceInterface";
-import { DiceTemplates } from "./DiceTemplates";
-
-export const diceCategories = [
-  faFireAlt,
-  faBalanceScale,
-  faShieldAlt,
-  faMagic,
-  faDragon,
-  faBroom,
-];
+import { DiceTemplates, GeneratedDiceTemplates } from "./DiceTemplates";
+import { Popover } from "./Popover";
+import { Button } from "./ui/Button";
+import { SmartTextInput } from "./ui/TextInput";
 
 export const DicePanel = React.memo(function DicePanel() {
-  const [active, setActive] = useState(0);
+  const [active, setActive] = useState("Input");
```

Comment:
This could probably use a better type than string, i.e. `useState<"Input" | "Skills" | "STs" | DiceTemplateCategoryID>`